src/zighttp/response.zig

```
@@ -1,3 +1,127 @@
 pub const Response = struct {
+    version: http.Version,
+    statusCode: i32,
+    statusText: []u8,
+    headers: std.StringHashMap([]const u8),
+    body: []u8,
+    allocator: std.mem.Allocator,
 
-};
\ No newline at end of file
+    pub fn init(allocator: std.mem.Allocator) !Response {
+        return Response{
+            .version = http.Version.HTTP_1_1,
+            .statusCode = 200,
+            .statusText = try allocator.dupe(u8, "OK"),
+            .headers = std.StringHashMap([]const u8).init(allocator),
+            .body = try allocator.dupe(u8, ""),
+            .allocator = allocator,
+        };
+    }
+
+    pub fn deinit(self: *Response) void {
+        self.allocator.free(self.statusText);
+
+        self.allocator.free(self.body);
+
+        var iter = self.headers.iterator();
+        while (iter.next()) |entry| {
+            self.allocator.free(entry.key_ptr.*);
+            self.allocator.free(entry.value_ptr.*);
+        }
+        self.headers.deinit();
+    }
+
+    pub fn setVersion(self: *Response, version: http.Version) void {
+        if (version == http.Version.HTTP_1_1)
+            self.version = version;
+    }
```

As we don't support other versions, we should throw an error when its not http/1.1